Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):<br>
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],<br>
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].<br>
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy<br>
Полученные значения должны быть равны.<br>
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,<br>
а затем с использованием функций из библиотек numpy и pandas.

Ковариацию можно вычислить по формуле: $cov_{XY}=M(XY)-M(X)M(Y)$

In [35]:
zp=[35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]
zpks=list(map(lambda x, y: x * y, zp, ks))
cov=sum(zpks)/len(zpks)-sum(zp)/len(zp)*sum(ks)/len(ks)
cov

9157.839999999982

In [36]:
import numpy as np
np.cov(zp, ks, ddof=0)


array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

коэффициент корреляции Пирсона: $r_{XY}=\frac{cov_{XY}}{\sigma_X\sigma_Y}$

In [37]:
mean_x=sum(zp)/len(zp)
mean_y=sum(ks)/len(ks)
Sx=(sum(list(map(lambda x:(x-mean_x)**2, zp)))/len(zp))**0.5
Sx

59.115480206118605

In [38]:
Sy=(sum(list(map(lambda x:(x-mean_y)**2, ks)))/len(ks))**0.5
Sy

174.55340157098058

In [39]:
r_xy=cov/(Sx*Sy)
r_xy

0.8874900920739144

In [40]:
np.std(zp, ddof=0)

59.115480206118605

In [41]:
np.std(ks, ddof=0)

174.55340157098058

In [43]:
np.cov(zp, ks, ddof=0)/(np.std(zp, ddof=0)*np.std(ks, ddof=0))

array([[0.33866702, 0.88749009],
       [0.88749009, 2.95275283]])

2. Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Поскольку в задаче имеется выборка малого объема при неизвестном значении среднеквадратического отклонения, применим распределение Стьюдента.<br>
$\overline{X}\pm t_{\frac{\alpha}{2}}\frac{\sigma}{\sqrt{n}}$<br>
n=10, $\alpha=0.5, \overline{X}$ - выборочное среднее.<br>
Величину $t_{\alpha,n-1}=t_{0.025,9}$ находим по таблицам распределения Стьюдента. Она равна 2.262.

In [45]:
x=[131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
np.mean(x)-2.262*np.std(x, ddof=1)/len(x)**0.5

110.55660776308164

In [46]:
x=[131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
np.mean(x)+2.262*np.std(x, ddof=1)/len(x)**0.5

125.64339223691834

Ответ: доверительный интервал [110.56; 125.64] для математического ожидания с надежностью 0.95.

3. Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

$\sigma=5$<br>
$\overline{X}=174.2$<br>
n=27<br>
$\alpha=5$<br>
$\overline{X}\pm Z_{\frac{\alpha}{2}}\frac{\sigma}{\sqrt{n}}=174.2\pm (1.96\frac{5}{\sqrt{27}})$<br>
$[\,172.314;176.086]\,$